docs/testplan/SmartSwitch-PMON-testplan.md

```diff
@@ -0,0 +1,939 @@
+#### Pass/Fail Criteria
+ *  The test result is a pass if the return value matches the expected value as shown in the "steps" and "Sample Output".
+
+### 1.5 Check SwitchModule APIs for SmartSwitch
+* This includes the appropriate new and modified ChassisClass and ModuleClass APIs 
+
+#### Steps
+ * get_dpu_id(self, name):
+    * Provide “SWITCH”
+    * Should return 0
+ * is_smartswitch(self):
+    * This API should return True
+ * get_module_dpu_port(self, index):
+    * For index 0 should return None
+ * get_base_mac(self):
+    * Should return the base mac address of the switch
+ * get_system_eeprom_info(self):
+    * Verify the returned dictionary key:value
+ * get_name(self):
+    * Verify if this API returns “SWITCH”
+ * get_description(self):
+    * Should return a string
+ * get_type(self):
+    * Should return “SWITCH” which is “MODULE_TYPE_SWITCH”
+ * get_oper_status(self):
+    * Should return the operational status of the DPU
+    * Stop one ore more containers
+    * Execute the CLI and see if it is down
+ * reboot(self, reboot_type):
+    * Issue this CLI with input “MODULE_REBOOT_DEFAULT”
+    * verify if the chassis is rebooted
+ * get_midplane_ip(self):
+    * should return the midplane IP of the switch
+
+#### Verify in
+ * Switch
+   
+#### Sample Output
+```
+On Switch:
+    get_dpu_id(self, SWITCH)
+    Output: 0
+
+    is_smartswitch(self):
+    Output: True
+
+    get_module_dpu_data_port(self, DPU0):
+    Output: None
+
+    get_base_mac(self):
+    Output: AA:CE:DD:D0:D0:78
+
+    get_system_eeprom_info(self):
+    Output: eeprom info object
+
+    get_name(self):
+    Output: SWITCH
+
+    get_description(self):
+    Output "Cisco 28x400G QSFPDD DPU-Enabled 2RU Smart Switch,Open SW"
+
+    get_type(self):
+    Output: SWITCH
+
+    get_oper_status(self):
+    Output: Online
+
+    reboot(self, reboot_type):
+    Result: the Chassis should reboot
+
+    get_midplane_ip(self):
+    Output: 169.254.200.254
+
+```
+#### Pass/Fail Criteria
+ *  The test result is a pass if the return value matches the expected value as shown in the "steps" and "Sample Output".
+ 
+ 
+### 1.6 Check the show reboot-cause CLI on the DPU
+
+#### Steps
+ * reboot the DPU
+ * issue the CLI “show reboot-cause” on the DPU
+ * verify the returned content’s time and cause
+ 
+#### Verify in
+ * DPU
+   
+#### Sample Output
+```
+On DPU: “show reboot-cause”
+Unknown
+```
+#### Pass/Fail Criteria
+ *  A proper output should be delayed without any error
+
+### 1.7 Check the show reboot-cause history CLI on the DPU
+
+#### Steps
+ * reboot the DPU a few times
+ * issue the CLI “show reboot-cause history” on the DPU
+ * verify the returned contents show all the reboot events time and cause
+ 
+#### Verify in
+ * DPU
+   
+#### Sample Output
+```
+On DPU: “show reboot-cause history”
+
+Output:
+
+Name                 Cause    Time    User    Comment
+-------------------  -------  ------  ------  ---------
+2024_06_04_23_27_28  Unknown  N/A     N/A     N/A
```

Comment:
Are Time and User fields populated for other reboot causes?

Reply:
Also, good to populate time for "unknown" cause as well.

Reply:
ok